WooCommerce/WooCommerceTests/ViewRelated/Blaze/BlazeCampaignCreationCoordinatorTests.swift

```diff
@@ -0,0 +1,148 @@
+        let featureFlagService = MockFeatureFlagService(blazei3NativeCampaignCreation: true)
+        insertProduct(.fake().copy(siteID: 1,
+                                   productID: 1,
+                                   statusKey: (ProductStatus.published.rawValue),
+                                   purchasable: true))
+
+        let sut = BlazeCampaignCreationCoordinator(siteID: 1,
+                                                   siteURL: "https://woo.com/",
+                                                   source: .campaignList,
+                                                   storageManager: storageManager,
+                                                   featureFlagService: featureFlagService,
+                                                   navigationController: navigationController,
+                                                   onCampaignCreated: { }
+        )
+        // When
+        sut.start()
+
+        // Then
+        XCTAssertTrue(navigationController.shownViewControllers[0] is BlazeCampaignCreationFormHostingController)
+    }
+
+    func test_given_enabled_i3_featureflag_when_no_product_id_supplied_and_there_are_multiple_eligible_products_then_navigate_to_product_selector() {
+        // Given
+        let navigationController = MockSourceNavigationController()
+        let featureFlagService = MockFeatureFlagService(blazei3NativeCampaignCreation: true)
+        insertProduct(.fake().copy(siteID: 1,
+                                   productID: 1,
+                                   statusKey: (ProductStatus.published.rawValue)))
+
+        insertProduct(.fake().copy(siteID: 1,
+                                   productID: 2,
+                                   statusKey: (ProductStatus.published.rawValue)))
+
+        let sut = BlazeCampaignCreationCoordinator(siteID: 1,
+                                                   siteURL: "https://woo.com/",
+                                                   source: .campaignList,
+                                                   storageManager: storageManager,
+                                                   featureFlagService: featureFlagService,
+                                                   navigationController: navigationController,
+                                                   onCampaignCreated: { }
+        )
+
+        // When
+        sut.start()
+
+        // Then
+        // The Product Selector is shown using a WooNavigationController
+        XCTAssertTrue(navigationController.presentedViewControllers[0] is WooNavigationController)
+
+        // The Product Selector is the first view controller in the presented WooNavigationController
+        let presentedNavigationController = navigationController.presentedViewControllers[0] as? WooNavigationController
+        let viewController = presentedNavigationController?.viewControllers[0]
+        XCTAssertTrue(viewController is ProductSelectorViewController)
```

Comment:
Nit: This can be simplified by using `try XCTUnwrap`:
```suggestion
        // The Product Selector is the first view controller in the presented WooNavigationController
        let presentedNavigationController = try XCTUnwrap(navigationController.presentedViewControllers.first as? WooNavigationController)
        let viewController = try XCTUnwrap(presentedNavigationController.viewControllers.first)
        XCTAssertTrue(viewController is ProductSelectorViewController)
```